source/Octopus.Server.Client/Operations/RegisterMachineOperation.cs

```diff
@@ -130,12 +138,11 @@ MachineResource GetMachine(IOctopusSpaceRepository repository)
             try
             {
                 existing = repository.Machines.FindByName(MachineName);
-                if (!AllowOverwrite && existing?.Id != null)
-                    throw new InvalidRegistrationArgumentsException($"A machine named '{MachineName}' already exists on the Octopus Server in the target space. Use the 'force' parameter if you intended to update the existing machine.");
```

Comment:
This exception message is different to the one below that I removed. Not sure why as they are the async/non-async versions of the same method. 🤔 

I've opted for the async version as it seems to be the one that is used.